### ISSUE TITLE
`status -n` and `status --current-line-number` not useful and underspecified

### ISSUE THREAD

**Issue by krader1961** [fish-shell/fish-shell#4161]
The documentation says `status --current-line-number` "prints the line number of the currently running script." While working on fixing issue #1310 I noticed this is a) not very useful and b) the implementation is wrong. I cannot find a single use of this feature in fish, Fisherman plugins, or OMF plugins other than my recently introduced `fish_breakpoint_prompt` function.

The zsh documentation is impenetrable (like so much about zsh behavior). The bash documentation is somewhat better:

> LINENO Each time this parameter is referenced, the shell substitutes a decimal number representing the current sequential line number (starting with 1) within a script or function.  When not in a script or function, the  value  substituted  is not guaranteed to be meaningful.  If LINENO is unset, it loses its special properties, even if it is subsequently reset.

If you have used `status current-line-number` (or `status -n`) please comment on why you used it and whether you found it useful.

We need to change how this feature works but before I do that I would like feedback from people on how they think it should work.

**Comment by zanchey** [comment]
Is it worth posting this on the mailing list as well?

**Comment by krader1961** [comment]
Note that `status --current-line-number` always returns a value relative to the start of the file. This means that if you define a function interactively the line number is relative to the `function xyz` statement. But if you put the definition in a file and `function xyz` isn't the first line of the file then the line number is no longer relative to the start of the function. It is relative to the start of the file contents.

What we need is to line numbers. One that is relative to the start of the function and one that is relative to the start of the file. The two will be the same if the function was defined interactively or `function xyz` is the first line of the file.

Command substitutions are also slightly weird. If you instrument the code you'll see that a statement like

```
set x (status current-line-number)
```

Always reports `1` regardless of where that statement appears in the function or file.

**Comment by krader1961** [comment]
While working on making `status --current-line-number`, etc., better defined and more useful I noticed we have exactly one unit test for this set of features:

```
# Test status -n
eval 'status -n
status -n
status -n'
```

It expects the output to be

```
1
2
3
```

Personally I don't find that very useful. It would be more useful to list the line number of the `eval` within the broader context of the function or file containing that statement.

**Comment by faho** [comment]
Fixed by 662fb3f3d1f81a05af60a7dab02bdc29152099c4 for now.
